File: src/main/java/corp/soyummy/userfavorites/UserFavorite.java
package corp.soyummy.userfavorites;

import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Field;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "userfavorites")
public class UserFavorite {
    @Id
    private String id;

    @Field("recipe")
    private String recipeId;

    @Field("userId")
    private String userId;
}

File: src/main/kotlin/com/soyummy/userfavorites/UserFavorite.kt
package com.soyummy.userfavorites

import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Field

@Document(collection = "userfavorites")
data class UserFavorite(
    @Id
    val id: String? = null,

    @Field("recipe")
    val recipeId: String,

    @Field("userId")
    val userId: String
)